src/clojush/random.clj

```diff
@@ -38,50 +38,84 @@
         (recur (inc parens)
                (rest probabilities))))))
 
+(defn conditional-thread
+  "Takes a value and threads it through the functions. If a function
+   returns nil, the old value will be the value passed to the next
+   function."
+  [val & fs]
+  (reduce #(or (%2 %1) %1) val fs))
+
+;; Example usage of conditional-thread (the function)
+;; (conditional-thread 0 inc inc inc)
+;; => 3
+;; (conditional-thread 0 #(when (= 0 %) 2)
+;;                        inc
+;;                        inc)
+;; => 4
+;; (conditional-thread 0 #(when (= 1 %) 2)
+;;                        inc
+;;                        inc)
+;; => 2
+
 (defn random-plush-instruction-map
   "Returns a random instruction map given the atom-generators and the required
    epigenetic-markers."
   ([atom-generators]
-    (random-plush-instruction-map atom-generators {}))
-  ([atom-generators {:keys [epigenetic-markers
-                            close-parens-probabilities
-                            silent-instruction-probability]
-                     :or {epigenetic-markers []
-                          close-parens-probabilities [0.772 0.206 0.021 0.001]
-                          silent-instruction-probability 0}}]
-    (let [markers (conj epigenetic-markers :instruction)]
-      (zipmap markers
-              (map (fn [marker]
-                     (case marker
-                       :instruction (let [element (lrand-nth atom-generators)]
-                                      (if (fn? element)
-                                        (let [fn-element (element)]
-                                          (if (fn? fn-element)
-                                            (fn-element)
-                                            fn-element))
-                                        element))
-                       :close (random-closes close-parens-probabilities)
-                       :silent (if (< (lrand) silent-instruction-probability)
-                                 true
-                                 false)
-                       ))
-                   markers)))))
+   (random-plush-instruction-map atom-generators {}))
+  ([atom-generators argmap]
+   (random-plush-instruction-map atom-generators false argmap))
```

Comment:
Again, the floating constant `false` is hard to interpret here.

Also a lot of this diff appears to be white space. Can we reformat so it matches the original code so it's easier to see what's really different.